syntax = "proto3";

import "graph.proto";
 
package transport_catalogue_protobuf;
 
message RoutingSettings {
    double bus_wait_time = 1;
    double bus_velocity = 2;
}

// edge_id_to_edge_

message BusEdge {
	string bus_name = 1;
	uint32 span_count = 2;
	double time = 3;
}

message StopEdge {
	string stop_name = 1; 
	double time = 2;  
}

message EdgeIdToEdge {
	uint32 edge_id  = 1;   
    oneof prev_edge {
		BusEdge bus_edge = 2;
		StopEdge stop_edge = 3;
	} 
}

// stop_to_router_

message RouterByStop{
	uint32 bus_wait_start = 1;
	uint32 bus_wait_end = 2;
}

message StopToRouter {
	uint32 stop_id = 1; 
	RouterByStop router_by_stop = 2;
}

message Weight {
	double weight = 1;
}

message PrevEdge {
	uint32 prev_edge = 1;
}

message RouteInternalData {
	Weight weight = 1;
	PrevEdge prev_edge = 2;
}

message RoutesInternalData {
    repeated RouteInternalData route_internal_data = 1;
}


message TransportRouterData {
	RoutingSettings settings = 1;
	repeated StopToRouter stop_to_router = 2;
	repeated EdgeIdToEdge edge_id_to_edge = 3;
	Graph graph = 4;
	repeated RoutesInternalData routes_inrernal_data = 5;

}